src/common/AdaptiveStream.cpp

```diff
@@ -268,15 +268,13 @@ bool AdaptiveStream::prepareDownload(const AdaptiveTree::Segment *seg)
           download_url_ = current_rep_->url_ + download_url_;
       }
       else
-      {
         download_url_ = current_rep_->url_;
-        uint64_t fileOffset = seg != &current_rep_->initialization_ ? m_segmentFileOffset : 0;
-        if (~seg->range_end_)
-          sprintf(rangebuf, "bytes=%" PRIu64 "-%" PRIu64, seg->range_begin_ + fileOffset, seg->range_end_ + fileOffset);
-        else
-          sprintf(rangebuf, "bytes=%" PRIu64 "-", seg->range_begin_ + fileOffset);
-        rangeHeader = rangebuf;
-      }
+      uint64_t fileOffset = seg != &current_rep_->initialization_ ? m_segmentFileOffset : 0;
+      if (~seg->range_end_)
+        sprintf(rangebuf, "bytes=%" PRIu64 "-%" PRIu64, seg->range_begin_ + fileOffset, seg->range_end_ + fileOffset);
+      else
```

Comment:
This does not work for URLSegment representations with no byte range given. The complete block must be placed inside a condition if (~seg->range_begin_)

Reply:
@peak3d : For sure this could happen. I added the requested changes.